dowhy/causal_refuters/non_parametric_sensitivity_analyzer.py

```diff
@@ -0,0 +1,270 @@
+import logging
+
+import matplotlib.pyplot as plt
+import numpy as np
+import pandas as pd
+import scipy
+from econml.utilities import cross_product
+from sklearn.compose import ColumnTransformer, make_column_selector
+from sklearn.linear_model import Lasso, LinearRegression
+from sklearn.model_selection import KFold
+from sklearn.pipeline import Pipeline
+from sklearn.preprocessing import StandardScaler
+
+from dowhy.causal_refuters.partial_linear_sensitivity_analyzer import PartialLinearSensitivityAnalyzer
+from dowhy.causal_refuters.reisz import get_alpha_estimator
+from dowhy.utils.util import generate_moment_function, get_generic_regressor, get_numeric_features
+
+
+class NonParametricSensitivityAnalyzer(PartialLinearSensitivityAnalyzer):
+    """
+    Non-parametric sensitivity analysis for causal estimators.
+
+    Currently works only for single-dimensional treatment and binary treatment because it uses a plugin estimate of reisz function.
+
+    TODO: Extend to general treatments by implementing the loss-based reisz representer.
+
+    Two important quantities used to estimate the bias are alpha and g.
+         g := E[Y | T, W, Z] denotes the long regression function
+         g_s := E[Y | T, W] denotes the short regression function
+         α := (T - E[T | W, Z] ) / (E(T - E[T | W, Z]) ^ 2) denotes long reisz representer
+         α_s := (T - E[T | W] ) / (E(T - E[T | W]) ^ 2) denotes short reisz representer
+    Bias = E(g_s - g)(α_s - α)
+    Thus, The bound is the product of additional variations that omitted confounders generate in the regression function and in the reisz representer for partially linear models.
+    It can be written as,  Bias = S * Cg * Calpha
+    where Cg and Calpha are explanatory powers of the confounder and S^2 = E(Y - g_s) ^ 2 * E(α_s ^ 2)
+
+    Based on this work:
+        Chernozhukov, Victor, Carlos Cinelli, Whitney Newey, Amit Sharma, and Vasilis Syrgkanis. Long Story Short: Omitted Variable Bias in Causal Machine Learning. No. w30302. National Bureau of Economic Research, 2022.

```

Comment:
I like the practice of citing the work in the code.  We should consider making that a more frequent practice when we are adding or updating algorithmic implementations.

Totally minor thing:  Victor's name is LAST, FIRST, but everyone else is FIRST LAST.  Intentional?

Reply:
ah, that was the Chicago format. I've changed it to APA.